### DISCUSSION TITLE
emitTo an extending component

### DISCUSSION THREAD

**Discussion by nuernbergerA** [livewire/livewire#5191] [4. General]
Given we have a Notification class

```php
class Notification extendes Component {
$listeners = ['some-event'];
}
```

And a variation of it

```php
class Alert extendes Notification {
//
}
```

If I would now call an another component ```$this->emitTo(Notification::class, 'some-event');```

Should Alert listen for that?

In my opinion, yes.
Currently it doesn't.

What are you thoughts?

**Comment by driesvints** [comment]
No, I don't feel it makes sense to apply a cascading effect here. You emit to a specific type of component. I don't feel like it's expected that other components receive the same event. How would Livewire know btw to which inheriting components to emit the event to?

If you cascade the event emitting to inheriting components then you can't undo that effect while now the other way around you can explicitly list all the component types you want to emit to.

**Reply by joshhanley** [reply]
On one hand I’d agree, it’d say it is a notification class and it’s listening for that event so should receive it, and I can see benefit in it being that way. 

But overall I think I agree with @driesvints and being explicit about which components you are emitting to is more flexible. If it cascaded you wouldn’t be able to stop it happening if you didn’t want it too.

**Reply by nuernbergerA** [reply]
Thanks @driesvints and @joshhanley for your opinion will stick to `emit` then
